### PR TITLE
Enh/refactor for clarity

### PR DESCRIPTION
**Pull request by BardiyaAk**
* Refactoring the code for clarity and adding hyperparameters to the output of the main functions.
* Add more error-handling to the code for making sets and creating prompts
* Add more helper functions to the sampler class to store information as a string (later we can use this to return this to users who are using setlexsem as a package)

### CODE REVIEW

File: setlexsem/generate/generate_sets.py [hunk 1]
```diff
@@ -45,19 +49,23 @@ def make_sets_from_sampler(
...
-        A, B = sample_set()
-        # loop through operations (on the same random sets)
-        set_list.append(
-            {
-                "experiment_run": i,
-                "A": A,
-                "B": B,
-            }
-        )
+        try:
```
Comment (phenomenal-manish) [inline setlexsem/generate/generate_sets.py:56]: Do we know when does sample_set() throws an exception apart from when it's the base class instance. We can add a unit test as well for failure when sampling.

Comment (BardiyaAk) [reply setlexsem/generate/generate_sets.py:56]: I like this, can you propose a unit test for this? It fails when the generator is empty. If there's a way to check for generator output without emptying it out, let's use that logic.

File: README.md [hunk 2]
```diff
@@ -4,12 +4,23 @@ This research repository maintains the code and the results for the research pap
 
 _"Set theory has become the standard foundation for mathematics, as every mathematical object can be viewed as a set."_ -[Stanford Encyclopedia of Philosophy](https://plato.stanford.edu/entries/set-theory/)
 
-# Install Package from PyPI and Use
+# TL;DR Install Package from PyPI and Use
```
Comment (ndronen) [inline README.md:7]: No, `tl;dr` is used informally at the beginning of a very long text. It's not really appropriate here.

Comment (ndronen) [reply README.md:7]: These are not titles: `# Install package from PyPI`.

Move the code example to another section.

File: README.md [hunk 3]
```diff
@@ -39,31 +50,39 @@ out_data = make_sets(
...
-    prompt_approach=["baseline", "chain_of_thought"],
-    is_fix_shot=[True],
+    prompt_config=prompt_config,
     add_roles=False)
 ```
 
 
 
-# Install
+# Development
```
Comment (ndronen) [inline README.md:86]: So now we have a section called `Development` that starts by talking about installing?

File: setlexsem/generate/generate_prompts.py [hunk 4]
```diff
@@ -328,7 +339,7 @@ def main(config_file, save_data, overwrite):
 
     # go through hyperparameters and run the experiment
     counter_exp = 1
-    for hp in make_hps_generator:
+    for hp_set in make_hps_generator:
```
Comment (ndronen) [inline setlexsem/generate/generate_prompts.py:342]: Start to use `hyperparam`, please.

**Review by ndronen — COMMENTED** [review]
LGTM. Will these changes make it easier to create single command that will just generate the dataset from the paper, so we don't need these fairly unhelpful code examples in the README?

**Review by ndronen — APPROVED** [review]

**Review by phenomenal-manish — APPROVED** [review]
Approving with note: I can take up refactoring sampling related code and adding unit tests as discussed in the comment thread.

**Pull request merged**